lldb/include/lldb/Target/ThreadPlanStepOverRange.h

```diff
@@ -42,8 +45,13 @@ class ThreadPlanStepOverRange : public ThreadPlanStepRange,
 
   void SetupAvoidNoDebug(LazyBool step_out_avoids_code_without_debug_info);
   bool IsEquivalentContext(const SymbolContext &context);
+  // Clear and create a new ThreadPlanSingleThreadTimeout to detect if program
```

Comment:
ThreadPlanStepOverRange isn't the only thread plan that we want to have this behavior for.  For instance, it would be great to do this for ThreadPlanStepOut, which we don't even try to run on one thread, IIRC, and in order not to duplicate code, we will want to switch the ThreadPlanCallUserExpression over to this method as well.  So the code to handle the interrupt has to be shareable - maybe a mix-in class that does the management of the interrupt?